src/OrchardCore/OrchardCore.Deployment.Core/DeploymentSourceSortedList.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace OrchardCore.Deployment.Core
+{
+    internal class DeploymentSourceSortedList : IEnumerable<IDeploymentSource>
```

Comment:
You don't need this class. The comparer can just cast a source to IOrderedDeploymentSource to order compared to another one. And call IEnumerable.OrderBy(comparer).

Reply:
Seb but I don't think there's a straightforward way to compare `IDeploymentSource` & `IOrderedDeploymentSource`. Am I right?